src/mapper.rs

```diff
@@ -0,0 +1,224 @@
+                    original,
+                    obfuscated,
+                    original_class,
+                    line_mapping,
+                    ..
+                } => {
+                    // in case the mapping has no line records, we use `0` here.
+                    let (startline, endline) =
+                        line_mapping.as_ref().map_or((0, 0), |line_mapping| {
+                            (line_mapping.startline, line_mapping.endline)
+                        });
+                    let (original_startline, original_endline) =
+                        line_mapping.map_or((0, None), |line_mapping| {
+                            match line_mapping.original_startline {
+                                Some(original_startline) => {
+                                    (original_startline, line_mapping.original_endline)
+                                }
+                                None => (line_mapping.startline, Some(line_mapping.endline)),
+                            }
+                        });
+                    let members = class.members.entry(obfuscated).or_insert_with(|| vec![]);
+                    members.push(MemberMapping {
+                        startline,
+                        endline,
+                        original_class,
+                        original,
+                        original_startline,
+                        original_endline,
+                    });
+                }
+                _ => {}
+            }
+        }
+        if !class.original.is_empty() {
+            classes.insert(class.obfuscated, class);
+        }
+
+        Self { classes }
+    }
+
+    /// Remaps an obfuscated Class.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use proguard::ProguardMapper;
+    ///
+    /// let mapping = r#"android.arch.core.executor.ArchTaskExecutor -> a.a.a.a.c:"#;
+    /// let mapper = ProguardMapper::from(mapping);
+    ///
+    /// let mapped = mapper.remap_class("a.a.a.a.c");
+    /// assert_eq!(mapped, Some("android.arch.core.executor.ArchTaskExecutor"));
+    /// ```
+    pub fn remap_class(&'s self, class: &str) -> Option<&'s str> {
+        self.classes.get(class).map(|class| class.original)
+    }
+
+    /// Remaps a single Stackframe.
+    ///
+    /// Returns zero or more [`StackFrame`]s, based on the information in
+    /// the proguard mapping. This can return more than one frame in the case
+    /// of inlined functions. In that case, frames are sorted top to bottom.
+    ///
+    /// [`StackFrame`]: struct.StackFrame.html
+    pub fn remap_frame(&'s self, frame: &StackFrame<'s>) -> RemappedFrameIter<'s> {
+        if let Some(class) = self.classes.get(frame.class.as_ref()) {
+            if let Some(members) = class.members.get(frame.method.as_ref()) {
+                let mut frame = frame.clone();
+                frame.class = class.original.into();
+                return RemappedFrameIter::members(frame, members.iter().fuse());
```

Comment:
The fuse here is redundant since `slice::Iter` [is already fused](https://doc.rust-lang.org/stable/std/slice/struct.Iter.html#impl-FusedIterator).